src/wp-admin/includes/class-wp-upgrader.php

```diff
@@ -737,7 +750,13 @@ public function run( $options ) {
 		 * Download the package (Note, This just returns the filename
 		 * of the file if the package is a local file)
 		 */
-		$download = $this->download_package( $options['package'], true );
+		$download = $this->download_package(
+			$options['package'],
+			true,
+			array(
```

Comment:
If the added third parameter is just `$options['hook_extra']` (not wrapped in an array), then the `wp_parse_args()` call in `download_package()` can be eliminated. The `hooks_extra` key will always be present in `run()` because there is a `wp_parse_args()` call in there.

Reply:
My initial thinking was that making the third arg `$hook_extra` might be restrictive in the future if someone wanted to add more parameters to the method. In hindsight this seems unlikely for this method. I've gone ahead and flattened everything 😄